KalturaMediaServer/src/main/java/com/kaltura/media/server/managers/KalturaLiveManager.java

```diff
@@ -219,15 +246,18 @@ public synchronized void unregister() {
 				timer = null;
 			}
 			registered = false;
+			readyForPlayback = false;
 
 			unsetEntryMediaServer(liveEntry, tmpIndex);
 		}
 
 		public boolean isRegistered() {
-			if (!registered)
-				return false;
+			return registered;
+		}
 
-			return registerTime.before(new Date());
+		public boolean isReadyForPlayback() {
+
+			return registered && readyForPlayback;
```

Comment:
1. u declared isRegistered 2 lines above this function, y did u not use it?
2. I dont see the logical connection between having someone registered & a ready stream